File: BE/models/Buy.js
const mongoose = require('mongoose');
const BuyDetails = new mongoose.Schema({
  dateOfBuy: Date,
  owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  },
  products: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Product'
  }],
  dateOfComming: Date
});
module.exports = mongoose.model("Buy", BuyDetails);

File: BE/modals/Buy.py
from mongoengine import Document, DateTimeField, ReferenceField, ListField, ObjectIdField

class Buy(Document):
    dateOfBuy = DateTimeField()
    owner = ReferenceField('User')  # הפניה למודל User
    products = ListField(ReferenceField('Product'))  # רשימת הפניות למודל Product
    dateOfComming = DateTimeField()